plugin/pkg/admission/gc/gc_admission.go

```diff
@@ -61,11 +66,32 @@ func (a *gcPermissionsEnforcement) Admit(attributes admission.Attributes) (err e
 		Path:            "",
 	}
 	allowed, reason, err := a.authorizer.Authorize(deleteAttributes)
-	if allowed {
-		return nil
+	if !allowed {
+		return admission.NewForbidden(attributes, fmt.Errorf("cannot set an ownerRef on a resource you can't delete: %v, %v", reason, err))
+	}
+
+	// Further check if the user is setting ownerReference.blockOwnerDeletion.
+	// If so, only allows the change if the user has delete permission of
+	// the _OWNER_
+	changedRefs := changingBlockOwnerDeletion(attributes.GetObject(), attributes.GetOldObject())
+	for _, ref := range changedRefs {
+		attribute, err := a.toDeleteAttribute(ref, attributes)
+		if err != nil {
+			// An error occurs if a RESTMapping cannot be found for
```

Comment:
Is this a security issue? If the RESTMapper is not up-to-date, we will allow `ObjectReferences` which would otherwise be forbidden. Until we have watchable discovery this might be fine as this only applies to TPRs. Can you add a comment that we switch to safe default "Forbidden" when we have watchable discovery ?